libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+
+This section may be omitted.
+
+### Parameters
+
+Parameters should take the form of a markdown unordered list. Each list item should start with the name of the parameter wrapped in backticks, followed by a brief, one-line description.
+
+The parameters in the list should be ordered exactly as they are in the function signature, and no parameter should be omitted.
+
+Each list item should read as a complete sentence.
+
+_Example:_
+
+```
+// - `tables` is a stream of tables
+// - `method` is the method to use when joining
+// - `on` is a list of column names on which to join
+```
+
+### Code Examples
+
+Code examples consist of a heading, an optional preamble, and a markdown-formatted code block.
+
+The heading should be a markdown H1 heading, and should be immediately followed by a blank line comment.
+
+Any text between the heading and the code block will be treated as part of the preamble. The preamble should be separated from the code block by a blank line comment. The preamble may be omitted.
+
+Code blocks should contain valid flux code that readers can copy and paste into the InfluxDB data explorer and run with no modifications. Any sample data used in a code example should come from an `array.from()` call in the example code itself, rather than being represented as a markdown- or html-formatted table.
+
+This section may include any number of code examples.
+
+_Example:_
+
+```
+// # Joining two tables
+// 
+// Run this in the data explorer to see the results!
+//
+// ```
+// import "array"
+//
+// sf_temp = array.from(
+//     rows: [
+//         {_time: 2021-06-01T01:00:00Z, _field: "temp", _value: 70},
+//         {_time: 2021-06-01T02:00:00Z, _field: "temp", _value: 75},
+//         {_time: 2021-06-01T03:00:00Z, _field: "temp", _value: 72},
+//     ],
+// )
+//
+// ny_temp = array.from(
+//     rows: [
+//         {_time: 2021-06-01T01:00:00Z, _field: "temp", _value: 55},
+//         {_time: 2021-06-01T02:00:00Z, _field: "temp", _value: 56},
+//         {_time: 2021-06-01T03:00:00Z, _field: "temp", _value: 57},
+//     ],
+// )
+//
+// join(
+//   tables: {sf: sf_temp, ny: ny_temp},
+//   on: ["_time", "_field"]
+// )
+// ```
+```
+
+### Other Information
+
+There are other details about Flux functions that we will want to document, like the return type, the types of the parameters, and whether or not a parameter is required. We should be able to get all of that information from the function signature, and therefore shouldn't need to include it in the doc comment.
+
+### Full Example
+
+For some examples of what a complete doc comment looks like for a flux function, see the doc comments for `join()` and `aggregateWindow()` in the `universe` package.
+
+## Documenting Packages
+
+### Import path
+
+Doc comments for packages should start with the full import path for the package, wrapped in backticks. Nothing else should be included on this line.
+
+_Example:_
+
+```
+// `experimental/geo`
```

Comment:
Can you explain the reason for this? My thought is that it would be easy to infer the package name without requiring an explicit doc comment.

Reply:
Hmmm... not sure why I included that, in retrospect. I'll remove it.